pages/_app/index.jsx

```diff
@@ -84,6 +84,10 @@ class App extends NextApp {
       // Don't report to console
       event.preventDefault()
     }
+
+    // Forcing reload because Next.js subscribes to this even first
+    if (event.reason instanceof NotFoundError)
```

Comment:
This sadly is very generic and may have an unintended effect.  You should probably create new exception like `DataProviderNotFound` or why don't you make changeDataProvider async so it can be caught correctly?

Reply:
I am not sure, making something to wait for this method is a good idea.

The flow should be: start changing, check conditions, create an event (error/success).

Reply:
That's fine. So in that case, if you can please create a new exception and don't rely on NotFoundError because it can be thrown by other API call too.

Reply:
Yes, I will

Reply:
@Joozty do you think we need a specific error or generic one is enough as it is now: yet another NotFoundError but from the store, not from the API?

Currently, it may be too generic since it's triggered only in a single specific case when the data provider is not found.

Reply:
I would say it would be better if it is specific something like `DataProviderNotFound`